xqueue.lua

```diff
@@ -394,7 +394,14 @@ function M.upgrade(space,opts,depth)
 	do
 		for _,index in pairs(space.index) do
 			if type(_) == 'number' then
-				if index.parts[1].fieldno == fields.status
+				if fields.priority then
+					if index.parts[1].fieldno == fields.status
+					and index.parts[2].fieldno == fields.priority
+					and index.parts[3].fieldno == self.key.no then
+						self.index = index
+						break
+					end
+				elseif index.parts[1].fieldno == fields.status
```

Comment:
It is better to use `else` here and nested `if` check inside. Also we have to check that `#index.parts >= 2`.